integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/ChangeDataTypeTestCases.scala

```diff
@@ -173,6 +175,48 @@ class ChangeDataTypeTestCases extends QueryTest with BeforeAndAfterAll {
     test_change_data_type()
   }
 
+  test("test alter change datatype for complex types") {
+    sql("drop table if exists test_rename")
+    sql("CREATE TABLE test_rename (name string) STORED AS carbondata")
+    // add complex columns
+    sql("alter table test_rename add columns(mapField1 MAP<int, int>, " +
+        "strField1 struct<a:int,b:decimal(5,2)>, arrField1 array<int>)")
+    sql("insert into test_rename values('df',map(5, 6),named_struct('a',1,'b', 123.45),array(1))")
+    // change datatype operation
+    sql("alter table test_rename change mapField1 mapField1 MAP<int, long>")
+    sql("alter table test_rename change strField1 strField1 struct<a:long,b:decimal(6,2)>")
```

Comment:
please once verify decimal type with lower to higher precision and scale change and vice versa

Reply:
Done, edited existing test case.